### PR TITLE
Make the ASE calculator warn for high uncertainties

### PR DESCRIPTION
**Pull request by frostedoyster**
Adds uncertainty warnings to the ASE calculator. The threshold is set to 0.1 eV/atom, which is "chemical accuracy", according to some. The tests need https://github.com/metatensor/lj-test/pull/5 to work correctly

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - ~Issue referenced (for PRs that solve an issue)?~

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

### CODE REVIEW

File: python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py [hunk 1]
```diff
@@ -321,6 +325,8 @@ def calculate(
         with record_function("ASECalculator::prepare_inputs"):
             outputs = _ase_properties_to_metatensor_outputs(properties)
             outputs.update(self._additional_output_requests)
+            if calculate_energy and self._model_has_energy_uncertainty:
+                outputs["energy_uncertainty"] = _get_energy_uncertainty_output()
```
Comment (Luthaf) [inline python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py:329]: Is this something we always want to do? I see a point for having it on by default, but I feel like users should be able to turn this off

File: python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py [hunk 2]
```diff
@@ -709,3 +729,12 @@ def _full_3x3_to_voigt_6_stress(stress):
             (stress[0, 1] + stress[1, 0]) / 2.0,
         ]
     )
+
+
+def _get_energy_uncertainty_output():
+    return ModelOutput(
+        quantity="energy",
```
Comment (Luthaf) [inline python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py:736]:
```suggestion
        quantity="energy_uncertainty",
```

File: python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py [hunk 3]
```diff
@@ -397,6 +403,20 @@ def calculate(
...
+            if calculate_energy and self._model_has_energy_uncertainty:
+                uncertainty = outputs["energy_uncertainty"].block().values
+                assert uncertainty.shape == (1, 1)
+                uncertainty = uncertainty.detach().cpu().numpy()[0, 0]
+                if uncertainty / len(atoms) > 0.1:
+                    # 100 meV/atom, which is more than 2 kcal/mol, means that this
+                    # prediction is not chemically accurate
+                    warnings.warn(
+                        "the model's energy uncertainty is larger than 100 meV/atom; "
+                        "this prediction is not chemically accurate",
```
Comment (Luthaf) [inline python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py:416]:
```suggestion
                        "the model's energy uncertainty is larger than 100 meV/atom ({uncertainty / len(atoms):.4} meV/atom), "
                        "you might be using it on a structure it can not handle",
```

File: python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py (hunk 1 above)
Comment (ceriottm) [reply python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py:329]: I tend to agree, it's good to have it on-by-default, but it will have a non-zero overhead so maybe one should be able to switch it off

File: python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py (hunk 3 above)
Comment (ceriottm) [reply python/metatensor_torch/metatensor/torch/atomistic/ase_calculator.py:416]: Should we make this threshold an option, and say that setting it to zero disables the check?
Also, IDK how hard it would be to get per-atom energies and uncertainties, but I think in many cases it would be a better check than looking at the mean uncertainty.

**Comment by Luthaf** [comment]
Can you re-open this in https://github.com/metatensor/metatomic?
